skpro/regression/gam.py

```diff
@@ -0,0 +1,262 @@
+# copyright: skpro developers, BSD-3-Clause License (see LICENSE file)
+"""Generalized Additive Models (GAM) Regressor."""
+
+__author__ = ["Omswastik-11", "dswah"]
+
+import numpy as np
+import pandas as pd
+
+# from skpro.distributions.inversegaussian import InverseGaussian
+from skpro.distributions.binomial import Binomial
+from skpro.distributions.gamma import Gamma
+from skpro.distributions.normal import Normal
+from skpro.distributions.poisson import Poisson
+from skpro.regression.base import BaseProbaRegressor
+
+
+class GAMRegressor(BaseProbaRegressor):
+    """Generalized Additive Model (GAM) Regressor.
+
+    Wraps the `pygam` library.
+
+    Parameters
+    ----------
+    terms : expression specifying terms to model, optional (default='auto')
+        By default a univariate spline term will be allocated for each feature.
+    distribution : str or pygam.Distribution, optional (default='normal')
```

Comment:
Can I suggest to use the same strings as the names of the `skpro` distribution, for consistency?